internal/ingress/controller/template/template.go

```diff
@@ -1672,3 +1673,43 @@ func convertGoSliceIntoLuaTable(goSliceInterface interface{}, emptyStringAsNil b
 		return "", fmt.Errorf("could not process type: %s", kind)
 	}
 }
+
+// buildOriginRegex returns an origin as a regex
+func buildOriginRegex(origin string) string {
+	if !strings.HasPrefix(origin, "*") {
+		return origin
+	}
+
+	origin = strings.Replace(origin, "*.", "", 1)
+	parts := strings.Split(origin, ".")
+
+	return "(" + strings.Join(parts, ".") + ")"
+}
+
+// buildCorsOriginRegex builds the regex string required by nginx
+func buildCorsOriginRegex(origins string) string {
+	var originSplitList = strings.Split(origins, ",")
+	if searchCorsOrigins(originSplitList, "*") {
+		return "set $http_origin *;\nset $cors 'true';"
+	}
+
+	var originsRegex string = "if ($http_origin ~* ("
+	for i, origin := range originSplitList {
```

Comment:
Since `origins` is a user provided string, we may want to handle empty strings (or strings just with spaces) in the split as well. Consider origins like `*.google.com,`.

This results in an empty string in the regex and I’m not sure if nginx will allow it to match all origins. Haven’t had a chance to try it out but just being paranoid

Reply:
i added a trimspace right after :) good catch